client/state/data-layer/wpcom-http/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { get, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import { WPCOM_HTTP_REQUEST } from 'state/action-types';
+import { extendAction } from 'state/utils';
+
+/**
+ * Returns the appropriate fetcher in wpcom given the request method
+ *
+ * fetcherMap :: String -> (Params -> Query -> [Body] -> Promise)
+ *
+ * @param {string} method name of HTTP method for request
+ * @returns {Function} the fetcher
+ */
+const fetcherMap = method => get( {
```

Comment:
this is unfortunate on several levels. I purposefully didn't add these two because the WordPress.com API doesn't currently allow anything but `POST` and `GET` requests and `wpcom.js` actually just aliasing `req.put` _and_ `req.del` as `req.post`.

It would bring me joy if we opened up the other methods because we could then talk about upholding common expectations about idempotency, cachability, and standard compliance in our API but for the meantime I'd rather acknowledge the reality of what's actually happening.

this coding shift should push more developers to the actual API docs/code and so I think that should be clear there.

in fact, I'm curious why parts of `wpcom.js` use `put` and `del` since it's lying about supporting those.

@retrofox any thoughts?